src/main.ts

```diff
@@ -21,4 +49,133 @@ async function run() {
   }
 }
 
+async function handleMain(inputs: Inputs, client: GitHubClient): Promise<void> {
+  const changes = await getChangesInPush();
+
+  if (changes.additions.includes(inputs.haltFile)) {
+    core.startGroup(`${inputs.defaultBranch}:${inputs.haltFile} added`);
+    const message = haltMessage(fs.readFileSync(inputs.haltFile).toString());
+    core.info(`Halting all open PRs: ${message}`);
+    await haltOpenPullRequests(client, github.context, inputs, message);
+    core.endGroup();
+  }
+
+  if (changes.removals.includes(inputs.haltFile)) {
+    core.startGroup(`${inputs.defaultBranch}:${inputs.haltFile} removed`);
+    core.info("Un-halting all open PRs");
+    await unhaltOpenPullRequests(client, github.context, inputs);
+    core.endGroup();
+  }
+}
+
+async function handlePullRequest(
+  inputs: Inputs,
+  client: GitHubClient,
+  pullRequest: PullRequest
+): Promise<void> {
+  const haltFile = await githubApi.getRepositoryContent(client, {
+    ...github.context.repo,
+    path: inputs.haltFile,
+    ref: inputs.defaultBranch,
+  });
+
+  const haltFileContents =
+    haltFile && "content" in haltFile ? atob(haltFile.content) : null;
```

Comment:
What is `&& "content"` doing here? Is this a thinko?

Reply:
This was an annoying yak AJ helped me through...

The content API returns `{ content: ... }` if you requested a file, or `[{ content:  }]` if you requested a directory, or (apparently) some other shapes for cases I don't understand.

The boolean expression `"content" in haltFile` is TypeScript for: "is this an object with a `.content` property?" and is the only way AJ and I could convince TypeScript I can call `.content` in the consequent path. The `haltFile &&` just catches the file not existing.

Altogether, it's:

```
const haltFileContents =
  { file existed && was a normal file } ? { Base64-decode the .content member } : null
```

Reply:
Does `hashOwnProperty` not work?

Reply:
Correct, it did not. We tried that first.